From 9d120163992f1ac2e38aec61a64a719a3636f698 Mon Sep 17 00:00:00 2001
From: Yanling Song <songyl@ramaxel.com>
Date: Fri, 26 Nov 2021 16:26:55 +0800
Subject: [PATCH] net:spnic:Clear the flag of CSUM and TSO when VXLAN is not
 supported
Patch-mainline: NO, OTHERS
References: OLK-5.10


Ramaxel inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4J0GH
CVE: NA

----------------------------------------

When the firmware does not support VXLAN offload,
 clear the flag of CSUM and TS0.

Signed-off-by: Yanling Song <songyl@ramaxel.com>
Reviewed-by: Xu Yun<xuyun@ramaxel.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/net/ethernet/ramaxel/spnic/spnic_main.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/ramaxel/spnic/spnic_main.c b/drivers/net/ethernet/ramaxel/spnic/spnic_main.c
index fa57a912cd5e..f09f11488042 100644
--- a/drivers/net/ethernet/ramaxel/spnic/spnic_main.c
+++ b/drivers/net/ethernet/ramaxel/spnic/spnic_main.c
@@ -235,8 +235,11 @@ static void netdev_feature_init(struct net_device *netdev)
 
 	netdev->priv_flags |= IFF_UNICAST_FLT;
 
-	netdev->hw_enc_features |= dft_fts | cso_fts;
-	netdev->hw_enc_features |= tso_fts | NETIF_F_TSO_ECN;
+	netdev->hw_enc_features |= dft_fts;
+	if (SPNIC_SUPPORT_VXLAN_OFFLOAD(nic_dev->hwdev)) {
+		netdev->hw_enc_features |= cso_fts;
+		netdev->hw_enc_features |= tso_fts | NETIF_F_TSO_ECN;
+	}
 }
 
 static void init_intr_coal_param(struct spnic_nic_dev *nic_dev)
-- 
2.26.2

